superdesk/publish/publish_content.py

```diff
@@ -51,7 +51,7 @@ def publish():
 
 
 def get_queue_items():
-    lookup = {'state': 'pending', 'destination.delivery_type': {'$ne': 'pull'}}
+    lookup = {'state': {'$in': ['pending', 'in-progress']}, 'destination.delivery_type': {'$ne': 'pull'}}
```

Comment:
I know but I still preferred this option because patching the item in `publish_queue` could fail so that item won't be retried at all.
I'll try your suggestion as well, although this flow is quite prone to race conditions.